tests/cases/fourslash/javaScriptClass1.ts

```diff
@@ -0,0 +1,32 @@
+///<reference path="fourslash.ts" />
+
+// Classes have their shape inferred from assignments
+// to properties of 'this' in the constructor
+
+// @allowNonTsExtensions: true
+// @Filename: Foo.js
+//// class Foo {
+////    constructor() {
+////        this.bar = 'world';
+////        this.thing = 42;
+////        this.union = 'foo';

```

Comment:
Can we check if JsDoc works for the assignment member, e.g. `@type`, or even `@params` if the assignment is a method?